Commands/SitesMassWordPressUpdateCommand.php

```diff
@@ -20,850 +21,1021 @@
 
-    $yaml_settings = $this->validateYamlSettings( $yaml_settings );
+			// Set connection back to sftp.
+			if ( $mode == 'git' ) {
 
-    return $yaml_settings;
+				$mode = $this->setSiteConnectionMode( $name, $environ, 'sftp' );
 
-  }
+			}
 
-  private function yamlGetSiteArgs( $name, $yaml ){
+			$autoDeployTest = $this->auto_deploy( $name, 'test' );
 
-    if ( ! isset( $yaml['sites']['update'] ) ) return array();
+			$autoDeployLive = $this->auto_deploy( $name, 'live' );
 
-    $args = false;
+		}
 
-    foreach ($yaml['sites']['update'] as $key => $value) {
 
-      if ( $value['name'] === $name ) {
+		//	Completion message.
+		if ( ! $report ) {
 
-        $args = $value;
+			$this->log()->notice( 'Finished plugins updates for {environ} environment of {name} site.', array(
+				'environ' => $environ,
+				'name' => $name,
+			) );
 
-        if ( isset( $yaml['sites']['settings'] ) ) {
+		}
 
-          $args = array_merge( $yaml['sites']['settings'], $args );
 
-        }
+		//	Send slack message
+		if ( isset( $this->yaml_settings['slack_settings'] ) ) {
 
-      }
+			$date = new \DateTime();
 
-    }
+			$slack_notif_text = "";
+			$slack_notif_text .= "*Terminus update on " . $name . "*\n\n";
 
-    return $args;
+			if ( $update_report ) {
 
-  }
+				$slack_notif_text .= "```";
 
-  private function yamlGetSiteSetting( $name, $setting ){
+				foreach ( $update_report as $report ) {
 
-    if ( ! $this->yaml_settings ) return false;
+					$slack_notif_text .= $report . "\n";
 
-    $settings = $this->yamlGetSiteArgs( $name, $this->yaml_settings );
+				}
 
-    foreach ($settings as $key => $value) {
+				$slack_notif_text .= "```";
 
-      if ( $key === $setting ) {
+			}
 
-        return $value;
+			$slack = simple_slack( $this->slack_settings['url'], array(
+				'username'		=> $this->slack_settings['username'],
+				'channel'		=> $this->slack_settings['channel'],
+				'icon_emoji'	=> $this->slack_settings['icon_emoji'],
+				'text'			=> $slack_notif_text,
+				'mrkdwn'		=> true,
+				'attachments'	=> array( array(
+					'fallback'		=> 'Terminus MWU message.',
+					'color'			=> '#ffb305',
+					'author_name'	=> 'Terminus MWU',
+					'fields'		=> array(
+						array(
+							'title'		=> 'Dashboard',
+							'value'		=> 'https://dashboard.pantheon.io/sites/',
+							'short'		=> true
+						),
+						array(
+							'title'		=> 'Dev',
+							'value'		=> ( isset( $env_url['dev'] ) ? $env_url['dev'] : 'no link provided.' ),
+							'short'		=> true
+						),
+						array(
+							'title'		=> 'Test',
+							'value'		=> ( isset( $env_url['test'] ) ? $env_url['test'] : 'no link provided.' ),
+							'short'		=> true
+						),
+						array(
+							'title'		=> 'Live',
+							'value'		=> ( isset( $env_url['live'] ) ? $env_url['live'] : 'no link provided.' ),
+							'short'		=> true
+						),
+					),
+					'image_url'		=> 'http://my-website.com/path/to/image.jpg',
```

Comment:
![screenshot 2016-10-10 07 44 31](https://cloud.githubusercontent.com/assets/7840931/19228148/7ad50be6-8eec-11e6-81b3-114825ce5374.png)

Those keys are going to attach an image on the message. It's not showing anything at the moment because it's not a valid image url, I'm putting it there just for user's reference in case they want to add images to the update message.

If they do, here's what it's going to looks like:

**For thumb_url**
https://api.slack.com/docs/messages/builder?msg=%7B%22text%22%3A%22I%20am%20a%20test%20message%20http%3A%2F%2Fslack.com%22%2C%22attachments%22%3A%5B%7B%22text%22%3A%22And%20here%27s%20an%20attachment!%22%2C%22thumb_url%22%3A%22https%3A%2F%2Fupload.wikimedia.org%2Fwikipedia%2Fmediawiki%2Fa%2Fa9%2FExample.jpg%22%7D%5D%7D

**For image_url**
https://api.slack.com/docs/messages/builder?msg=%7B%22text%22%3A%22I%20am%20a%20test%20message%20http%3A%2F%2Fslack.com%22%2C%22attachments%22%3A%5B%7B%22text%22%3A%22And%20here%27s%20an%20attachment!%22%2C%22image_url%22%3A%22https%3A%2F%2Fupload.wikimedia.org%2Fwikipedia%2Fmediawiki%2Fa%2Fa9%2FExample.jpg%22%7D%5D%7D